docs/python-integration.md

```diff
@@ -0,0 +1,299 @@
+        ),
+        target_schema="public",
+        target_table="blog_embeddings",
+        view_schema="public",
+        view_name="blog_embeddings_view"
+    )
+```
+
+# Alembic Integration 
+
+To actually create the vectorizer, pgai provides two alembic helpers:
+
+## Creating a Vectorizer
+
+Basic creation:
+
+```python
+from alembic import op
+from pgai.configuration import EmbeddingConfig, ChunkingConfig
+
+def upgrade():
+    op.create_vectorizer(
+        'blog_posts',
+        embedding=EmbeddingConfig(
+            model='text-embedding-3-small',
+            dimensions=768
+        ),
+        chunking=ChunkingConfig(
+            chunk_column='content',
+            chunk_size=700
+        ),
+        formatting_template='Title: ${title}\nContent: ${chunk}'
+    )
+```
+
+## Dropping a Vectorizer
+
+```python
+def downgrade():
+    # Drop by ID
+    op.drop_vectorizer(1, drop_all=True)
+```
+
+The `drop_all=True` parameter will also clean up the associated embedding table and view.
+
+## Autogeneration Support
+
+If you don't want to write the configuration twice, you can make use of alembics autogenerate feature to automatically detect changes between your SQL models and the underlying database schema.
+
+### Setup
+
+To configure autogeneration, you need to import a custom comparison function as well as exclude pgai managed models from alembics usual comparators via the `include_object` parameter:
+
+```python
+from alembic import context
+from pgai.alembic import compare_vectorizers
+from pgai.alembic import CreateVectorizerOp, DropVectorizerOp
+
+
+# Make sure your env.py includes:
+def run_migrations_online():
+    with connectable.connect() as connection:
+        context.configure(
+            connection=connection,
+            target_metadata=target_metadata,
+            include_object=lambda obj, name, type_, reflected, compare_to:
+            not obj.info.get("pgai_managed", False)
+        )
+```
+
+### Features
+
+The autogeneration system will:
+
+1. Detect new vectorizers defined in models and generate creation operations
+2. Detect removed vectorizers and generate drop operations
+3. Detect changes in vectorizer configuration and generate update operations (as drop + create)
+
+Note: All operations done are **not reversible** via `alembic downgrade`.
+
+Example generated migration for a new vectorizer:
+
+```python
+def upgrade():
+    op.create_vectorizer(
+        'blog_posts',
+        embedding=EmbeddingConfig(
+            model='text-embedding-3-small',
+            dimensions=768
+        ),
+        chunking=ChunkingConfig(
+            chunk_column='content',
+            chunk_size=500
+        )
+    )
+
+def downgrade():
+    op.drop_vectorizer(1)
+```
+
+When changes are made to a model's vectorizer configuration, the autogeneration will create appropriate migration operations:
+
+```python
+def upgrade():
+    # Update vectorizer configuration
+    op.drop_vectorizer(1, drop_objects=True)
```

Comment:
We should probably make it more clear this is a dangerous and destructive operations which requires re-embedding everything again and paying $$. Perhaps just a code comment explaining this is sufficient.